scipy/linalg/tests/test_lapack.py

```diff
@@ -1666,75 +1666,90 @@ def generate_random_dtype_array(shape, dtype):
+    - sva matches singular values of A
+    - info == 0
+    - that sva, v, and u can be recombined to make A
+    - that u.T @ u and v.T @ v are the identity
+    Lastly, we check that iwork [0] and [1] are the matrix rank.
     '''
-    np.random.seed(42)
-    rtol = 250 * np.finfo(dtype).eps
+    seed(42)
     atol = 100 * np.finfo(dtype).eps
     m, n = size
     A = generate_random_dtype_array((m, n), dtype)
-
     gejsv = get_lapack_funcs('gejsv', dtype=dtype)
-    # pass A and gejsv into lambda expression for parametrized job? values
-    sva, u, v, work, iwork, info = gejsv_lambda(A, gejsv)
+
+    jobs = {}
+
+    if job is None:
+        # use default gejsv values
+        sva, u, v, work, iwork, info = gejsv(A)
+    elif type(job) == list:
+        # execute with modified job argument
+        out = eval("gejsv(A, " + ",".join(list(map("=".join, job))) + ")")
+        sva, u, v, work, iwork, info = out
+        # add each job to dictionary
+        for job_ in job:
+            jobs[job_[0]] = job_[1]
+    else:
+        sva, u, v, work, iwork, info = eval("gejsv(A, " + "=".join(job) + ")")
+
+        jobs[job[0]] = job[1]
 
     assert_equal(info, 0)
 
     SIGMA = np.diag(work[1] / work[0] * sva[:n])
-    sigma = work[1] / work[0] * sva[:n] # not a matrix
+    sigma = work[1] / work[0] * sva[:n]
 
+    assert_allclose(np.sort(sigma), np.sort(svd(A)[0]), atol=atol)
 
-    assert_allclose(np.sort(sigma), np.sort(eig(A)[0]), rtol=rtol, atol=atol)
-    # ^ this doesn't seem like it would work as sva is real and eig(A) is complex
-    assert_allclose(A, u @ SIGMA @ v.T, rtol=rtol, atol=atol)
-    assert_allclose(u.T @ u, np.identity(u.shape[1]), rtol=rtol, atol=atol)
-    assert_allclose(v @ v.T, np.identity(n), rtol=rtol, atol=atol)
+    # run this test if jobu is computed
+    if not jobs.get('jobu') == "'N'":
+        assert_allclose(u.T @ u, np.identity(u.shape[1]), atol=atol)
 
-    #assert_equal(iwork[0], np.linagl.matrix_rank(A))
-    #assert_equal(iwork[1], np.linagl.matrix_rank(A))
+    # run this test if jobv is computed
+    if not jobs.get('jobv') == "'N'":
+        assert_allclose(v @ v.T, np.identity(n), atol=atol)
+
+    # run this if both jobu and jobv are computed
+    if (jobs.get('jobu') != "'N'" and
+            jobs.get('jobv') != "'N'"):
+        assert_allclose(A, u @ SIGMA @ v.T, atol=atol)
+
+    # check the rank of the matrix is correctly reported in iwork
+    assert_equal(iwork[0], np.linagl.matrix_rank(A))
+    assert_equal(iwork[1], np.linagl.matrix_rank(A))
 
 
 @pytest.mark.parametrize("dtype", DTYPES)
 def test_gejsv_specific(dtype):
     '''
-    Tests more specific cases of routine ?gejsv, specifically for when
-    u and v are not computed, matrix A is 1D, 1x1, empty, lwork = 0,
-    and invalid job? parameters.
+    Tests more specific cases of routine ?gejsv:
+    - when matrix is not full rank
+    - matrix A is 1D, 1x1, empty
+    - lwork = 0
+    - invalid job? parameters.
```

Comment:
I saw in the documentation that for option `jobv='J'`

> This option is  allowed only if JOBU .NE. 'N', i.e. in computing the full SVD.

Should I include some sort of a check here just to see if gejsv fails or has non zero info if `jobu='N'`?

Reply:
Sure. We wouldn't want it to segfault, anyway. See what it does and test for that.